browser/todo-list.md

```diff
@@ -3,291 +3,248 @@
-| `components/Profile.js` | Application data storage and user sign out. |
+### Add, edit and delete todos privately
 
-<!-- The `src/components/App.js` code configures an `AppConfig` object and then uses this to create a `UserSession`. Then, the application calls a [`redirectToSignIn()`](https://blockstack.github.io/blockstack.js#redirectToSignIn) function which generates the `authRequest` and redirects the user to the Blockstack authenticator: -->
+Once you've authenticated the app, you can can start adding todos by entering values into the "Write your to do" field and hitting "Enter".
 
-The `src/components/App.js` code configures a `UserSession` and other `authOptions`, which are passed to the `Connect` component. The `Connect` component acts as a "provider" for the rest of your application, and essentially creates a re-usable configuration for you.
+![](images/todos-home-authenticated.svg)
 
-In the `src/components/Signin.js` component, we are then calling the `useConnect` hook. This hook returns many helper functions, one of which is `doOpenAuth`. Calling this method will being the authentication process. First, it injects a modal into your application, which acts as a way of "warming up" your user to Blockstack authentication. When the user continues, they are redirected to the Blockstack authenticator, where they can finish signing up.
+The data for all todos are saved as JSON to the Gaia hub linked to your Secret Key using the [`putFile`](http://blockstack.github.io/blockstack.js/globals.html#putfile) method of the `userSession` object in the [`src/assets/data-store.js`](https://github.com/blockstack/blockstack-todos/blob/master/src/assets/data-store.js#L26) module:
 
 ```js
-import React from 'react';
-import { useConnect } from '@blockstack/connect';
-
-export const Signin = () => {
-  const { doOpenAuth } = useConnect();
-
-  return (
-    <button
-      onClick={() => doOpenAuth()}
-    >
-      Sign In with Blockstack
-    </button>
-  )
+export const saveTasks = async (userSession, tasks, isPublic) => {
+  await userSession.putFile(TASKS_FILENAME, JSON.stringify({ tasks, isPublic }), {
+    encrypt: !isPublic,
+  });
 };
 ```
 
-Once the user authenticates, the application handles the `authResponse` in the `src/components/Profile.js` file. :
+These todos are subsequently loaded using the [`getFile`](http://blockstack.github.io/blockstack.js/globals.html#getfile) method of the same object in the same module:
 
 ```js
-...
-componentWillMount() {
-    if (userSession.isSignInPending()) {
-      userSession.handlePendingSignIn().then((userData) => {
-        //if (!userData.username) {
-        //  throw new Error('This app requires a username.')
-        //}
-        window.location = window.location.origin;
-      });
-    }
-  }
-...
+export const fetchTasks = async (userSession, username) => {
+  const tasksJSON = await userSession.getFile(TASKS_FILENAME, {
+    decrypt: false,
+    username: username || undefined,
```

Comment:
as a first time reader, I might ask why this is in there? how could the username be null? Maybe an explanation here could help?